packages/pantheon/src/index.ts

```diff
@@ -0,0 +1,177 @@
+  chiefSpouse?: string;
+  deities: Array<Deity>;
+  seed: string;
+}
+
+const createPantheon = (seed: string = sower.silly()): Pantheon => {
+  // seed our random number generator
+  mediator.provide('random', seedrandom(seed));
+
+  const deities = [];
+  const dispositions = make(3).map(() => generateDisposition());
+  const relationships = [];
+
+  // Step 1: Create a chief deity
+  const chief = createDeity(dispositions, []);
+  relationships.push({
+    first: 'anon-chronos',
+    second: chief.id,
+    action: 'beget',
+  });
+
+  // Step 2: Create a "sideboard" of deities. These are
+  // potential suitors for the second generation and potential
+  // relatives of the chief, children of titans and eldritch horrors.
+  const sideBoard = make(10).reduce((acc, _, idx) => {
+    const currentArchs = [chief.archetype, ...acc.map(({ archetype }) => archetype)];
+    const deity = createDeity(dispositions, currentArchs);
+
+    if (d(10) === 8) {
+      relationships.push({
+        first: d(2) === 1 ? 'anon-chronos' : 'anon-uranus',
+        second: deity.id,
+        action: 'beget',
+      });
+    }
+
+    acc.push(deity);
+    return acc;
+  }, []);
+
+  // Step 3. Determine if the chief should be matriarchal and self-procreating
+  // If the chief is a hermaphrodite, he will also be selfReproducing.
+  // Based on this, a spouse will either be created or the chief assigned
+  // as spouse.
+  const shouldBeMatriarch = chief.gender === 'female' && d(50) === 50;
+  const selfReproducing = shouldBeMatriarch || chief.gender === 'hermaphrodite';
+  const spouse = selfReproducing
+    ? { id: chief.id, gender: chief.gender, relationships: [] }
+    : createDeity(dispositions, []);
+
+  deities.push(...[chief, spouse, ...sideBoard]);
+
+  // Step 4. Generate consort relationships. These are randomly created, with the chief
+  // and chief's spouse being guaranteed. Anon are mortal (or animal) consorts
+  // that will produce demigods.
+  const consorts = [
+    {
+      first: chief.id,
+      second: spouse.id,
+      guarantee: 3,
+    },
+    ...make(8).map(() => ({
+      first: chief.id,
+      second: 'anon',
+    })),
+    ...(selfReproducing
+      ? []
+      : make(8).map(() => ({
+          first: spouse.id,
+          second: 'anon',
+        }))),
+  ];
+
+  const unpackGeneration = (acc, { relationships: generatedRelationships, children }) => {
+    if (!children.length) return acc;
+
+    relationships.push(...generatedRelationships);
+    acc.push(...children);
+    return acc;
+  };
+
+  // Step 5. Make babies
+  const secondGeneration = consorts
+    .map(
+      reproduce(
+        dispositions,
+        deities.map(({ archetype }) => archetype),
+      ),
+    )
+    .reduce(unpackGeneration, []);
+  deities.push(...secondGeneration);
+
+  const availableConsorts = deities.filter((d) => {
+    const hasTwoParents = relationships.filter(({ second }) => second === d.id);
+
+    return hasTwoParents.length === 2;
```

Comment:
nitpick: hasTwoParents sounds like it should itself be a bool, rather than an array. could chain .length to filter and name it parent count and the  === 2 would explain we're looking for deities with 2 parents

Reply:
Oh yea solid call.